specs/localdisco-v4.rst

```diff
@@ -0,0 +1,77 @@
+.. _localdisco-v3:
+
+Local Discovery Protocol v4
+===========================
+
+Mode of Operation
+-----------------
+
+Each participating device periodically sends an Announcement packet. It also
+keeps a table of the announcements it has seen. There is no way to solicit a
+reply; the only message type is Announcement.
+
+On multihomed hosts the announcement packets should be sent on each interface
+on which Syncthing will accept connections.
+
+The announcement packet is sent over UDP.
+
+For IPv4, the Announcement packet is broadcast either to the link-specific
+broadcast address, or to the generic link-local broadcast address
+``255.255.255.255``, with destination port 21027.
+
+For IPv6, the Announcement packet is multicast to the transient link-local
+multicast address ``ff12::8384``, with destination port 21027.
+
+It is recommended that local discovery Announcement packets be sent on a 30 to
+60 second interval, possibly with immediate transmissions when a previously
+unknown device is discovered.
+
+Device ID
+---------
+
+The device ID is the SHA-256 (32 bytes) of the device X.509 certificate. See
+:ref:`device-ids` in the Syncthing documentation.
+
+Announcement packet
+-------------------
+
+The Announcement packet has the following structure:
+
+.. code-block:: none
+
+     0                   1                   2                   3
+     0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    |                             Magic                             |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    /                                                               /
+    \                       Announce Message                        \
+    /                                                               /
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+There is no explicit length field as the length is given by the length of
+the discovery announcement packet itself.
+
+The Magic field is a 32 bit word representing 0x2EA7D90B in big endian byte
```

Comment:
I see the network byte order call out early on in the BEP spec. It just so happens that the BEP spec appears earlier in my github review than the discovery spec, but that is not necessarily globally true. I agree that once at the top of each spec is sufficient.

I was ambiguous in "I like the BEP spec way better, personally". What I meant was that there was a difference between the way local discovery and BEP described byte order. I want to make them consistent. They could be converging on BEP's style or LD's style. I like the (BEP spec way) better. I was not saying 'I like the (BEP spec) (way better).'